### DISCUSSION TITLE
Brainstorming for additional functionalities and features

### DISCUSSION THREAD

**Discussion by JonasSchaub** [Steinbeck-Lab/ChemIcal-DatasEt-comparatoR#16] [Ideas]
Here, we can collect and discuss ideas for new features and functionalities. 

Notes from our document so far:
- Functional group frequency analysis based on Ertl alg. - which functional groups appear and how frequently? Grid image of the most frequent ones
- Give CIDER a user interface (web app?)
- More from[ https://github.com/PatWalters/practical_cheminformatics_tutorials](https://github.com/PatWalters/practical_cheminformatics_tutorials)?
- Additional inspiration from Molecule Set Comparator application?

**Comment by JonasSchaub** [comment]
Make other input file types available?

**Reply by OBrink** [reply]
I would definitely agree that we should add more input files. And although a web app might be a bit much, it would probably enable access for a lot more people. It does not need to be anything fancy, just some kind of user interface that runs the workflow

**Reply by adelenelai** [reply]
+1 for SMILES input

**Comment by JonasSchaub** [comment]
One idea I had: We could add initial structure curation functionalities to CIDER. But I would say only if it can be integrated easily from existing toolkits (like the RDKit ChEMBL curation pipeline, for example). Or we need to add an explicit statement that all curation must be done beforehand.

**Reply by JonasSchaub** [reply]
@hannbus @Kohulan @OBrink as discussed this morning. Anything to add?

**Comment by JonasSchaub** [comment]
Idea for a new feature: Generate plots like the ones in Figures 3 and 4 of this paper https://pubs.acs.org/doi/10.1021/acs.jnatprod.8b01022 for the scaffolds and the maybe upcoming functional group analysis (or any kind of fragment analysis).

<img width="320" alt="image" src="https://user-images.githubusercontent.com/44881147/203284082-d1116a7f-8832-43f8-8577-fa00538c3db3.png">

**Reply by OBrink** [reply]
But please without this horrible colour scheme! :D

**Comment by OBrink** [comment]
Parallelisation to speed things up would be useful

**Comment by JonasSchaub** [comment]
Clustering - e.g. https://rdkit.readthedocs.io/en/latest/Cookbook.html#clustering-molecules could result in a simple bar plot of the clusters and how many molecules are in each cluster, as a quick metric for diversity.

**Comment by JonasSchaub** [comment]
Wrap dictionary data structure in a class object?

**Comment by JonasSchaub** [comment]
New idea: Being able to set a "memory mode", i.e. keeping mol objects in memory (e.g. for servers) vs. re-importing the molecules every time (e.g. for laptops).

**Comment by JonasSchaub** [comment]
Chemical space visualisations with descriptor vectors as input, i.e. every position in the input vector is a specific molecular descriptor value. This is nice because it helps to interpret the principal components in some way based on how much each descriptor value contributes to the respective dimension (loading plot).

**Reply by JonasSchaub** [reply]
ChemPlot seems to be able to do this using the setting "tailored" instead of "structural".

**Comment by JonasSchaub** [comment]
https://www.mdpi.com/2218-273X/10/11/1566
See this paper for some more ideas
